src/null-safety/faq.md

```diff
@@ -319,72 +319,75 @@ and situations where a null value is really expected. So the tool tells you what
 it knows ("it looks like this condition will always be false!") and lets you
 decide what to do.
 
-## What should I know about dart2js and null safety?
+## What should I know about compiling and null safety?
 
-For a long time, the dart2js compiler has had optimizations specifically
-targeting null values and null checks. Because of that, null safety is not
-expected to affect much the output of the compiler.
+The compiler optimizes null values and null checks. So null safety
+should not affect the compiler output much.
 
 A few notes that are worth highlighting:
 
-* dart2js emits `!` null assertions, but you may not notice them.
-  That's because pre null-safety dart2js already emitted null checks
-  (so they are already in the existing programs).
-
-* dart2js emits these null assertions regardless of sound/unsound null safety
-  and regardless of optimization level. In fact, dart2js doesn't remove
-  `!` when using `-O3` or `--omit-implicit-checks`.
-
-* dart2js may optimize away unnecessary null checks. This is because the same
-  optimizations done by dart2js in the past will be able to eliminate
-  unnecessary null checks when it knows the value is not null.
-
-* By default dart2js would generate parameter subtype checks (runtime checks
-  used to ensure covariant virtual calls are given appropriate arguments).
-  These are elided with the `--omit-implicit-checks` flag just as before.
-  Recall that this flag can make programs have unexpected behavior if types
-  are invalid, so we continue to recommend that the code has strong test
-  coverage to avoid any surprises. In particular, dart2js optimizes code based
+* The compiler generates `!` null assertions, but you might not notice
+  them. Pre null-safety, the compiler generated null checks that still
```

Comment:
nit: reading this with fresh eyes - this is not very easy to follow. Maybe the subject here is to implicit (my fault when I first added this text, I think). Suggestion:

* The production web compile generates `!` null assertions, but you might not notice them when comparing the output of the compiler before and after adding null assertions. That's because the compiler already emitted null checks in programs that weren't null-safe.